components/core/src/Utils.cpp

```diff
@@ -213,3 +219,124 @@ ErrorCode read_list_of_paths (const string& list_path, vector<string>& paths) {
 
     return ErrorCode_Success;
 }
+
+void load_lexer_from_file (std::string schema_file_path,
+                           bool reverse,
+                           log_surgeon::lexers::ByteLexer& lexer) {
+    FileReader schema_reader;
+    schema_reader.try_open(schema_file_path);
+    /// TODO: this wrapper is repeated a lot
+    log_surgeon::Reader reader_wrapper{[&](char* buf, size_t count, size_t& read_to) -> log_surgeon::ErrorCode {
+        schema_reader.read(buf, count, read_to);
+        if (read_to == 0) {
+            return log_surgeon::ErrorCode::EndOfFile;
+        }
+        return log_surgeon::ErrorCode::Success;
+    }};
+    log_surgeon::SchemaParser sp;
+    std::unique_ptr<log_surgeon::SchemaAST> schema_ast = sp.generate_schema_ast(reader_wrapper);
+    auto* delimiters_ptr = dynamic_cast<log_surgeon::DelimiterStringAST*>(
+            schema_ast->m_delimiters.get());
+    if (!lexer.m_symbol_id.empty()) {
+        throw std::runtime_error("Error: symbol_ids initialized before setting enum symbol_ids");
+    }
+    /// TODO: this is a copy of other code
+    lexer.m_symbol_id[log_surgeon::cTokenEnd] = (int) log_surgeon::SymbolID::TokenEndID;
+    lexer.m_symbol_id[log_surgeon::cTokenUncaughtString] =
+            (int) log_surgeon::SymbolID::TokenUncaughtStringID;
+    lexer.m_symbol_id[log_surgeon::cTokenInt] = (int) log_surgeon::SymbolID::TokenIntId;
+    lexer.m_symbol_id[log_surgeon::cTokenFloat] = (int) log_surgeon::SymbolID::TokenFloatId;
+    lexer.m_symbol_id[log_surgeon::cTokenFirstTimestamp] = (int) log_surgeon::SymbolID::TokenFirstTimestampId;
+    lexer.m_symbol_id[log_surgeon::cTokenNewlineTimestamp] = (int) log_surgeon::SymbolID::TokenNewlineTimestampId;
+    lexer.m_symbol_id[log_surgeon::cTokenNewline] = (int) log_surgeon::SymbolID::TokenNewlineId;
+
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenEndID] = log_surgeon::cTokenEnd;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenUncaughtStringID] =
+            log_surgeon::cTokenUncaughtString;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenIntId] = log_surgeon::cTokenInt;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenFloatId] = log_surgeon::cTokenFloat;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenFirstTimestampId] =
+            log_surgeon::cTokenFirstTimestamp;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenNewlineTimestampId] =
+            log_surgeon::cTokenNewlineTimestamp;
+    lexer.m_id_symbol[(int) log_surgeon::SymbolID::TokenNewlineId] = log_surgeon::cTokenNewline;
+
+    /// TODO: figure out why this needs to be specially added
```

Comment:
elaborate on the problem

Reply:
Added description on difference between the log_surgeon::SymbolIDs and why some are not added, some are added explicitly, and some are added based on the user schema.